### PR TITLE
ISPN-6776 Remove the listener on the client side

### PR DESCRIPTION
**Pull request by karesti**
https://issues.jboss.org/browse/ISPN-6766

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerRemoveOnStopTest.java [hunk 1]
```diff
@@ -30,4 +32,24 @@ public void testAllListenersRemovedOnStop() {
...
 
+   public void testRemoveListenerAfterStopAndRestart() {
+      final RemoteCache<Integer, String> rcache = remoteCacheManager.getCache();
+      final EventLogListener<Integer> eventListener1 = new EventLogListener<>(rcache);
+      rcache.addClientListener(eventListener1);
+      Set<Object> listeners = rcache.getListeners();
+      assertEquals(1, listeners.size());
+      assertEquals(eventListener1, listeners.iterator().next());
+      listeners = rcache.getListeners();
+      assertEquals(1, listeners.size());
```
Comment (tristantarrant) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerRemoveOnStopTest.java]: Why are you checking twice ?

Comment (karesti) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerRemoveOnStopTest.java]: forfot to delete the lines ! test above have 2 listeners but for the test I've done proving the bug I only needed one ! I deleted the double check

**Comment by tristantarrant** [comment]
Ran it locally:
`
[TestSuiteProgress] Test starting: org.infinispan.client.hotrod.event.ClientListenerRemoveOnStopTest.testRemoveListenerAfterStopAndRestart
[TestSuiteProgress] Test failed: org.infinispan.client.hotrod.event.ClientListenerRemoveOnStopTest.testRemoveListenerAfterStopAndRestart
10:05:52,953 ERROR (TestNG) [TestSuiteProgress] Test failed: org.infinispan.client.hotrod.event.ClientListenerRemoveOnStopTest.testRemoveListenerAfterStopAndRestart
org.infinispan.client.hotrod.exceptions.RemoteCacheManagerNotStartedException:: Cannot perform operations on a cache associated with an unstarted RemoteCacheManager. Use RemoteCacheManager.start before using the remote cache.
	at org.infinispan.client.hotrod.impl.RemoteCacheImpl.assertRemoteCacheManagerIsStarted(RemoteCacheImpl.java:512)
	at org.infinispan.client.hotrod.impl.RemoteCacheImpl.addClientListener(RemoteCacheImpl.java:437)
	at org.infinispan.client.hotrod.event.ClientListenerRemoveOnStopTest.testRemoveListenerAfterStopAndRestart(ClientListenerRemoveOnStopTest.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)[TestSuiteProgress] Tests succeeded: 1, failed: 1, skipped: 0
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.243 sec <<< FAILURE! - in org.infinispan.client.hotrod.event.ClientListenerRemoveOnStopTest
`

**Comment by karesti** [comment]
@tristantarrant for a reason I had launched all the test suite together in my head, but I was just launching the second one, and the first one stops the remoteCacheManager so it is stopped after. 
This causes dependency between tests, is it working now for you ? didn't realize about the test failing !

**Comment by tristantarrant** [comment]
Pushed to master and 9.0.x.

**Comment by danberindei** [comment]
@karesti Wrong JIRA reference in the commit message :)

Also, although 50 chars (as [the post referenced by the contributing guide](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) recommends) seems pretty hard to achieve when we include the JIRA reference, you should try to make the first line of the commit message as short as possible.

**Comment by karesti** [comment]
@danberindei sorry !! I got it wrong from the creation branch in my fork with the jira number ! I duplicated the wrong number ! can we fix this in the git history or we leave it like that ? 
Ok for the message ! thank you for pointing me to the post

**Pull request merged**